fe/fe-core/src/main/java/com/starrocks/connector/iceberg/jdbc/IcebergJDBCCatalog.java

```diff
@@ -0,0 +1,209 @@
+import com.starrocks.connector.iceberg.IcebergCatalog;
+import com.starrocks.connector.iceberg.IcebergCatalogType;
+import com.starrocks.connector.iceberg.cost.IcebergMetricsReporter;
+import com.starrocks.connector.iceberg.io.IcebergCachingFileIO;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.aws.AwsProperties;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.jdbc.JdbcCatalog;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.starrocks.connector.ConnectorTableId.CONNECTOR_ID_GENERATOR;
+import static com.starrocks.connector.iceberg.IcebergConnector.ICEBERG_CUSTOM_PROPERTIES_PREFIX;
+import static com.starrocks.connector.iceberg.IcebergMetadata.LOCATION_PROPERTY;
+import static org.apache.iceberg.CatalogProperties.WAREHOUSE_LOCATION;
+
+public class IcebergJDBCCatalog implements IcebergCatalog {
+    private static final Logger LOG = LogManager.getLogger(IcebergJDBCCatalog.class);
+    private final Configuration conf;
+    private final JdbcCatalog delegate;
+
+    public IcebergJDBCCatalog(String name, Configuration conf, Map<String, String> properties) {
+        this.conf = conf;
+
+        Map<String, String> copiedProperties = Maps.newHashMap(properties);
+
+        properties.forEach((key, value) -> {
+            String newKey = key.toLowerCase();
+            if (newKey.startsWith(ICEBERG_CUSTOM_PROPERTIES_PREFIX)) {
+                newKey = newKey.substring(ICEBERG_CUSTOM_PROPERTIES_PREFIX.length());
+                copiedProperties.put(newKey, value);
+            }
+        });
+
+        copiedProperties.put(CatalogProperties.FILE_IO_IMPL, IcebergCachingFileIO.class.getName());
+        copiedProperties.put(CatalogProperties.METRICS_REPORTER_IMPL, IcebergMetricsReporter.class.getName());
+        copiedProperties.put(AwsProperties.CLIENT_FACTORY, IcebergAwsClientFactory.class.getName());
+
+        if (!copiedProperties.containsKey(WAREHOUSE_LOCATION)) {
+            throw new IllegalArgumentException(String.format("Iceberg JDBC catalog must set warehouse location (\"%s\" = \"s3://path/to/warehouse\").",
+                    ICEBERG_CUSTOM_PROPERTIES_PREFIX + WAREHOUSE_LOCATION));
+        }
+
+        delegate = (JdbcCatalog) CatalogUtil.loadCatalog(JdbcCatalog.class.getName(), name, copiedProperties, conf);
+    }
+
+    @Override
+    public IcebergCatalogType getIcebergCatalogType() {
```

Comment:
Do we need to help user to automatically create these metadata table according to https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java#L57